policy/modules/services/container.if

```diff
@@ -0,0 +1,840 @@
+## <summary>
+##	Domain allowed to transition.
+## </summary>
+## </param>
+#
+interface(`container_domtrans_podman_user',`
+	gen_require(`
+		type podman_user_t, podman_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, podman_exec_t, podman_user_t)
+')
+
+########################################
+## <summary>
+##	Execute podman in the podman user domain,
+##	and allow the specified role the container
+##	domain (rootless podman).
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition
+##	</summary>
+## </param>
+## <param name="role">
+##	<summary>
+##	The role to be allowed the container domain.
+##	</summary>
+## </param>
+#
+interface(`container_run_podman_user',`
+	gen_require(`
+		attribute_role container_roles;
+	')
+
+	container_domtrans_podman_user($1)
+	roleattribute $2 container_roles;
+')
+
+########################################
+## <summary>
+##	Send and receive messages from
+##	container engines over dbus.
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed access.
+## </summary>
+## </param>
+#
+interface(`container_engine_dbus_chat',`
+	gen_require(`
+		attribute container_engine_domain;
+		class dbus send_msg;
+	')
+
+	allow $1 container_engine_domain:dbus send_msg;
+	allow container_engine_domain $1:dbus send_msg;
+')
+
+########################################
+## <summary>
+##	All of the permissions necessary
+##	for a container engine to manage
+##	container processes.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`container_manage_process',`
+	gen_require(`
+		attribute container_domain;
+	')
+
+	allow $1 container_domain:process { getattr getsched setsched transition signal signull sigkill };
+')
+
+########################################
+## <summary>
+##	Allow the specified domain to
+##	perform a type transition to
+##	container domains.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition.
+##	</summary>
+## </param>
+#
+interface(`container_transition',`
+	gen_require(`
+		attribute container_domain;
+	')
+
+	allow $1 container_domain:process transition;
+')
+
+########################################
+## <summary>
+##	Connect to a container domain
+##	over a unix stream socket.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`container_stream_connect',`
+	gen_require(`
+		attribute container_domain;
+		type container_runtime_t;
+	')
+
+	files_search_runtime($1)
+	stream_connect_pattern($1, container_runtime_t, container_runtime_t, container_domain)
+')
+
+########################################
+## <summary>
+##	Allow the specified domain to
+##	send all signals to a container
+##	domain.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition.
+##	</summary>
+## </param>
+#
+interface(`container_signal_process',`
```

Comment:
Seems like it should be `container_signal_all()` or `container_signal_all_containers()`

Reply:
Looks like there is a similar naming problem for the above interfaces too.

Reply:
I think I like `container_signal_all_containers()`.